### PR TITLE
Converted functional tests to async model

### PR DESCRIPTION
**Pull request by xavierdecoster**
Also upgraded selenium dependencies to latest available.

### CODE REVIEW

**Comment by xavierdecoster** [comment]
Already used this branch to run the tests from our CI and verified they work as intended (they produce the same results as the current ones), but the code is a bit more maintainable :)

Please review @bhuvak @maartenba

File: tests/NuGetGallery.FunctionalTests.Fluent/AccountManagement/ContactEmailTest.cs [hunk 1]
```diff
@@ -100,7 +100,7 @@ public void ContactEmailSupport()
             // Wait for 30 secs. to make sure that the mail is delivered properly.
             received = false;
             for (int i = 0; ((i < 10) && !received); i++)
```
Comment (yishaigalatzer) [inline tests/NuGetGallery.FunctionalTests.Fluent/AccountManagement/ContactEmailTest.cs]: nit:  The extra parens are ugly an not necessary

File: tests/NuGetGallery.FunctionalTests.Fluent/PackageManagement/PackageManagementTest.cs [hunk 2]
```diff
@@ -25,8 +24,8 @@ public void PackageManagement()
 
             // List 1.0.0
             I.Open(UrlHelper.BaseUrl + "/packages/" + packageName + "/1.0.0/Delete");
-            listedCheckbox = I.Find("#Listed").Invoke() as FluentAutomation.Element;
-            if (!(listedCheckbox.Attributes.Get("checked") == "true"))
+            var listedCheckbox = I.Find("#Listed").Element;
+            if (listedCheckbox.Attributes.Get("checked") != "true")
```
Comment (yishaigalatzer) [inline tests/NuGetGallery.FunctionalTests.Fluent/PackageManagement/PackageManagementTest.cs]: :) :100:

File: tests/NuGetGallery.FunctionalTests.Helpers/AssertAndValidationHelper.cs [hunk 3]
```diff
@@ -153,35 +144,34 @@ public static WebTestRequest GetVerifyPackagePostRequestForPackage(WebTest test,
...
         /// </summary>
         /// <param name="packageId"></param>
         /// <param name="version"></param>
-        public static void UploadNewPackageAndVerify(string packageId, string version = "1.0.0", string minClientVersion = null, string title = null, string tags = null, string description = null, string licenseUrl = null, string dependencies = null)
+        /// <param name="minClientVersion"></param>
+        /// <param name="title"></param>
+        /// <param name="tags"></param>
+        /// <param name="description"></param>
+        /// <param name="licenseUrl"></param>
+        /// <param name="dependencies"></param>
```
Comment (yishaigalatzer) [inline tests/NuGetGallery.FunctionalTests.Helpers/AssertAndValidationHelper.cs]: just remove the unused xml docs

**Comment by yishaigalatzer** [comment]
:shipit:

**Comment by maartenba** [comment]
:shipit:, looks good

**Pull request merged**